### PR TITLE
Remove old save options and reorder them

### LINKED ISSUE #18334: Update ISIS SANS Save options
Steve mentined:

>NXcanSAS is now ratified by the NIAC and following review v1.0 has been tagged for release.
 >https://github.com/nexusformat/definitions/pull/516
 
>Please can you check Save/LoadNXcanSAS for compliance against this release revision in time for Mantid 3.9?
 
>There have been a couple of structural changes if I remember correctly, and some renaming of elements…
 
 
>Just to check: In 3.9, the ‘NeXus’ output format checkbox on the SANS GUI will change to ‘NXcanSAS’ (and point to SaveNXcanSAS)?
 
>Can we remove the ‘CSV’ option from the GUI at the same time, please (I’m uncomfortable encouraging the use of formats with no metadata any longer)?
 
 
>It would also probably make sense to re-sequence the format options from:
 
>NeXus     NIST Qxy     CanSAS     RKH     CSV
 
>to, I propose:
 
>NIST Qxy     RKH     NXcanSAS     CanSAS     
 
>because (L-R) the formats are then 2D only, 1D/2D, 1D/2D, and 1D only.

### PR DESCRIPTION
**Pull request by AntonPiccardoSelg**
Removes SaveCSV and SaveNexusProcessed and reorders the save options, as requested by Steve.

Fixes #18334

## To test

1. Open the ISIS SANS GUI
2. Select the LOQ instrument
3. Load the user file: USER_LOQ_161G_Hedstrom_8mm_Magnet-MERGED.txt
4. Set   LOQ97072.nxs as the sample scatter input
5. Press Load Data
6. Press 1D Reduce
7. In the save group box, check `RKH`, `NXcanSAS` adn `CanSAS`
8. Press `Save Result`
  * Confirm that the results are saved without errors (make sure that they were saved on your file system)
9. Click `Save Other` and save dialog will appear
10. Select `RKH`, `NXcanSAS` adn `CanSAS` as the algorithm option, select `97074main_1D_2.2_10.0` as the workspace to save by clicking on it
11. Press `Save`
  * Confirm that the results are saved without errors (make sure that they were saved on your file system)

## Release notes

See [here](https://github.com/mantidproject/mantid/pull/18337/commits/f125cda6e3bb978bdf52ccf180750567b8b97486)
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [x] Is the code of an acceptable quality?
- [x] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by AntonPiccardoSelg** [comment]
Test data:

[USER_LOQ_161G_Hedstrom_8mm_Magnet-MERGED.txt](https://github.com/mantidproject/mantid/files/668595/USER_LOQ_161G_Hedstrom_8mm_Magnet-MERGED.txt)
[FLAT_CELL_LOQ_MAIN_143.TXT](https://github.com/mantidproject/mantid/files/668596/FLAT_CELL_LOQ_MAIN_143.TXT)
[DIRECT_LOQ_MAIN_143B.txt](https://github.com/mantidproject/mantid/files/668597/DIRECT_LOQ_MAIN_143B.txt)
[DIRECT_LOQ_HAB_983.txt](https://github.com/mantidproject/mantid/files/668598/DIRECT_LOQ_HAB_983.txt)

File: docs/source/release/v3.9.0/sans.rst [hunk 1]
```diff
@@ -20,6 +20,7 @@ Bug Fixes
 - Fix for LARMOR multi-period loading. The initial rotation was not correctly applied to all child workspaces.
 - IDF bug when using Larmor in batch mode was resolved.
 - Issue where Gui changes were not picked up for batch reductions was resolved.
+- Remove SaveNexusProcessed and SaveCSV as an option. Reorder options by dimensionality.
```
Comment (ianbush) [inline docs/source/release/v3.9.0/sans.rst]: Should this mention that this is in the GUI?

**Comment by ianbush** [comment]
The changes all work as described. The only comment I have is that on Ubuntu 14.04 the save options labels are not fully visible (see screenshots below). It looks like there should be space to make them wider though.

stretched
default

**Comment by AntonPiccardoSelg** [comment]
@ianbush  The labels should display now without being cut off. The merge issue has also been fixed.

**Comment by ianbush** [comment]
Looks good :shipit: .

**Pull request merged**